source/simulator/solver.cc

```diff
@@ -231,28 +214,20 @@ namespace aspect
     BlockSchurPreconditioner (const LinearAlgebra::BlockSparseMatrix     &S,
                               const LinearAlgebra::BlockSparseMatrix     &Spre,
                               const SInvOperator                         &S_inv_operator,
-                              const AInvOperator                         &A_inv_operator,
-                              const bool                                  do_solve_A,
-                              const bool                                  A_block_symmetric,
-                              const double                                A_block_tolerance
-                             )
+                              const AInvOperator                         &A_inv_operator)
       :
       stokes_matrix     (S),
       stokes_preconditioner_matrix     (Spre),
       S_inv_operator(S_inv_operator),
-      A_inv_operator  (A_inv_operator),
-      do_solve_A        (do_solve_A),
-      A_block_is_symmetric(A_block_symmetric),
-      n_iterations_A_(0),
-      A_block_tolerance(A_block_tolerance)
+      A_inv_operator  (A_inv_operator)
     {}
 
     template <class AInvOperator, class SInvOperator>
     unsigned int
     BlockSchurPreconditioner<AInvOperator, SInvOperator>::
     n_iterations_A() const
     {
-      return n_iterations_A_;
+      return A_inv_operator.n_iterations;
     }
```

Comment:
Hm, this is unsatisfying. It requires you to look into the inner object. Is there no better way?

Reply:
You are of course correct.